Sources/PackageCollectionsSigning/Certificate/CertificatePolicy.swift

```diff
@@ -18,9 +30,300 @@ protocol CertificatePolicy {
+            guard try self.hasExtendedKeyUsage(.codeSigning, in: certChain[0]) else {
+                return self.callbackQueue.async { callback(.failure(CertificatePolicyError.codeSigningCertRequired)) }
+            }
+            // Must support OCSP
+            guard try self.supportsOCSP(certificate: certChain[0]) else {
+                return self.callbackQueue.async { callback(.failure(CertificatePolicyError.ocspSupportRequired)) }
+            }
+
+            // Verify the cert chain - if it is trusted then cert chain is valid
+            self.verify(certChain: certChain, anchorCerts: self.trustedRoots, diagnosticsEngine: self.diagnosticsEngine, callbackQueue: self.callbackQueue, callback: callback)
+        } catch {
+            return self.callbackQueue.async { callback(.failure(error)) }
+        }
+    }
+}
+
+/// Policy for validating developer.apple.com certificates.
+///
+/// This has the same requirements as `DefaultCertificatePolicy` plus additional
+/// marker extensions for Apple Distribution certifiications.
+struct AppleDeveloperCertificatePolicy: CertificatePolicy {
+    private static let expectedCertChainLength = 3
+    private static let appleDistributionIOSMarker = "1.2.840.113635.100.6.1.4"
+    private static let appleDistributionMacOSMarker = "1.2.840.113635.100.6.1.7"
+    private static let appleIntermediateMarker = "1.2.840.113635.100.6.2.1"
+
+    let trustedRoots: [Certificate]?
+    let expectedSubjectUserID: String?
+
+    private let callbackQueue: DispatchQueue
+    private let diagnosticsEngine: DiagnosticsEngine?
+
+    /// Initializes a `AppleDeveloperCertificatePolicy`.
+    /// - Parameters:
+    ///   - trustedRootCertsDir: On Apple platforms, all root certificates that come preinstalled with the OS are automatically trusted.
+    ///                          Users may specify additional certificates to trust by placing them in `trustedRootCertsDir` and
+    ///                          configure the signing tool or SwiftPM to use it. On non-Apple platforms, only trust root certificates in
+    ///                          `trustedRootCertsDir` are trusted.
+    ///   - expectedSubjectUserID: The subject user ID that must match if specified.
+    ///   - callbackQueue: The `DispatchQueue` to use for callbacks
+    ///   - diagnosticsEngine: The `DiagnosticsEngine` for emitting warnings and errors.
+    init(trustedRootCertsDir: URL? = nil, expectedSubjectUserID: String? = nil, callbackQueue: DispatchQueue = DispatchQueue.global(), diagnosticsEngine: DiagnosticsEngine? = nil) {
+        self.trustedRoots = trustedRootCertsDir.map { Self.loadCerts(at: $0, diagnosticsEngine: diagnosticsEngine) }
+        self.expectedSubjectUserID = expectedSubjectUserID
+        self.callbackQueue = callbackQueue
+        self.diagnosticsEngine = diagnosticsEngine
+    }
+
     func validate(certChain: [Certificate], callback: @escaping (Result<Void, Error>) -> Void) {
```

Comment:
nit: you can wrap the callback with the callbackQueue in the start of this function which will make the rest of code simpler